identity/src/rsa.rs

```diff
@@ -55,27 +58,48 @@ impl Keypair {
     /// format (i.e. unencrypted) as defined in [RFC5208].
     ///
     /// [RFC5208]: https://tools.ietf.org/html/rfc5208#section-5
-    pub fn try_decode_pkcs8(der: &mut [u8]) -> Result<Keypair, DecodingError> {
-        let kp = RsaKeyPair::from_pkcs8(der)
-            .map_err(|e| DecodingError::failed_to_parse("RSA PKCS#8 PrivateKeyInfo", e))?;
-        der.zeroize();
-        Ok(Keypair(Arc::new(kp)))
+    pub fn try_decode_pkcs8(bytes: &mut [u8]) -> Result<Keypair, DecodingError> {
+        match RsaKeyPair::from_pkcs8(bytes) {
+            Ok(kp) => {
+                let kp = Self {
+                    inner: Arc::new(kp),
+                    raw_key: bytes.to_vec(),
+                };
+                bytes.zeroize();
+                Ok(kp)
+            }
+            Err(e) => Err(DecodingError::failed_to_parse("RSA", e)),
+        }
```

Comment:
Any reason for the change in coding style? Why not stick to `?`?

Reply:
Happened when resolving conflicts. I'll fix that later.